alica_tests/src/test/test_config_change.cpp

```diff
@@ -18,7 +18,7 @@ TEST_F(AlicaNotInitialized, TestUpdatingComponents)
     alica::AlicaCreators creators(std::make_unique<alica::ConditionCreator>(), std::make_unique<alica::UtilityFunctionCreator>(),
             std::make_unique<alica::ConstraintCreator>(), std::make_unique<alica::BehaviourCreator>(), std::make_unique<alica::PlanCreator>());
 
-    EXPECT_TRUE(ac->setOption<bool>("Alica.SilentStart", false));
+    EXPECT_TRUE(ac->setOption<bool>("Alica.SilentStart", false)); // engine not yet created so AlicaContext::reload gives error
```

Comment:
but the engine is constructed in `AlicaTestNotInitializedFixture`. Is this test still failing?

Reply:
Yes, the setOption should be done before AlicaContext::init so I changed the fixture
also
this piece of code needs to be removed for me (the _engine doesn't still exist here):
```c++
AlicaContext::setOptions()
{
...
    if (reload) {
        reloadConfig();
    }
...
}
Finally, some tests need to be split.

Reply:
the changes you have now made look fine to me

Reply:
I have changed and now setOption can be executed also after AlicaContext::Init.